sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -2424,6 +2426,72 @@ private void SetNameFieldsIfNull()
                         operationName: "BlobBaseClient.GetProperties",
                         cancellationToken: cancellationToken)
                         .ConfigureAwait(false);
+                    if (response.Value.ObjectReplicationRuleStatus.Count > 0)
+                    {
+                        IDictionary<string, string> ParsedRuleIdStatus = new Dictionary<string, string>();
```

Comment:
I think we should put this logic in BlobExtensions.cs (might not exist in this branch, see https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeExtensions.cs as an example.

Reply:
Also, I think we should rename this ```ObjectReplicationRules```.